### PR TITLE
DEV-388: Cassandra module: Create tables using JSON schema, assign tables to commands group, insert command into multiple assigned tables

### PR DESCRIPTION
**Pull request by Merisho**
Introduce:

- Cassandra Storage Provider

- TableSchema and Query builders

### CODE REVIEW

File: test/cassandra/src/CassandraStorage.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        const cassandra = new CassandraStorage(MockCassandraClient, config);
+        cassandra.initializeTableSchemas(config.tables);
+        cassandra.assignTablesToCommands(...config.commandTables);
+
+        cassandra.insertCommand({});
+
+        assert.equal(batchSpy.firstCall.args[0].length, 1);
+    });
+
+    it('Should set null value in columns which are not defined as properties in command object', () => {
```
Comment (Merisho) [inline test/cassandra/src/CassandraStorage.js]: Do we really need this test? We can just omit properties that don't exist, Cassandra will put null for us during INSERT

File: test/cassandra/src/CassandraStorage.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+        cassandra.insertCommand(command);
+
+        const [ firstQueryInBatch ] = batchSpy.firstCall.args[0];
+        assert.equal(batchSpy.firstCall.args[0].length, 1);
+        assert.equal(firstQueryInBatch.query, 'INSERT INTO commands (command, timestamp) VALUES (?, ?)');
+        assert.deepEqual(firstQueryInBatch.params, [ 'command-name', 1516266743223 ]);
+    });
+
+    it('Should execute query only for tables that defined', () => {
```
Comment (Merisho) [inline test/cassandra/src/CassandraStorage.js]: Please specify in test name what kind of query it's going to be executed

File: test/cassandra/src/CassandraStorage.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        MockCassandraClient.prototype.execute = sinon.spy();
+        const batchSpy = MockCassandraClient.prototype.batch = sinon.spy();
+
+        const cassandra = new CassandraStorage(MockCassandraClient, config);
+        cassandra.initializeTableSchemas(config.tables);
+        cassandra.assignTablesToCommands(...config.commandTables);
+
+        cassandra.insertCommand(command);
+
+        assert.equal(batchSpy.firstCall.args[0].length, 2);
```
Comment (Merisho) [inline test/cassandra/src/CassandraStorage.js]: It's a bit unclear what `batchSpy.firstCall.args[0]` stands for, please use temporary explaining variable

File: test/cassandra/src/CassandraStorage.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
+const assert = require('assert');
+const sinon = require('sinon');
+
+const CassandraStorage = require('../../../cassandra/src/CassandraStorage');
+
+describe('Cassandra Storage Provider', () => {
+    it('Should call execution of table creation query for each described table', () => {
+        const config = {
```
Comment (Merisho) [inline test/cassandra/src/CassandraStorage.js]: Please introduce DataBuilder for JSON schema to use in tests if it's possible. A lot of initial data create mess in tests

File: cassandra/src/utils.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    prepareConfigPropertyName(propName) {
+        return propName.split('.').map(camelCase).join('.');
+    },
+
+    booleanOrDefault(str) {
+        if(str === 'true' || str === 'false') {
+            return str === 'true';
+        }
+
+        return str;
```
[the remark this answers is not in the document]
Comment (Merisho) [reply cassandra/src/utils.js]: I considered this solution but I try to avoid nested ternary operators, IMHO it looks a bit confusing.

File: cassandra/lib/TableSchemaBuilder.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    constructor() {
+        this._query = '';
+        this._table = '';
+        this._columnsWithKeys = '';
+        this._ifNotExists = '';
+    }
+
+    createTable(tableName = '') {
+        this._query = `CREATE TABLE`;
+        !!tableName && this.withName(tableName);
```
[the remark this answers is not in the document]
Comment (Merisho) [reply cassandra/lib/TableSchemaBuilder.js]: I am just wondering why it's not the best way in this case. Could you please give an example?

File: cassandra/lib/TableSchemaBuilder.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+    ifNotExists() {
+        this._ifNotExists = 'IF NOT EXISTS';
+        return this;
+    }
+
+    build() {
+        let query = this._query + ' ';
+        if(!!this._ifNotExists) {
+            query += this._ifNotExists + ' ';
```
Comment (Merisho) [inline cassandra/lib/TableSchemaBuilder.js]: We can move this concatenation to ifNotExists()

Comment (Merisho) [reply cassandra/lib/TableSchemaBuilder.js]: And try to avoid !! cast to boolean

File: cassandra/src/CassandraUtils.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    static prepareConfigPropertyName(propName) {
+        return propName.split('.').map(camelCase).join('.');
+    }
+
+    static booleanOrDefault(str) {
+        return ({ 'true': true, 'false': false })[str] || str;
+    }
+
+    static isObject(val) {
+        return val && typeof val === 'object';
```
Comment (Merisho) [inline cassandra/src/CassandraUtils.js]: Use already created and well-tested solutions in such cases, for this particularly you can take lodash implementation of isObject
